packages/common-all/src/dnode.ts

```diff
@@ -882,10 +895,7 @@ export class NoteUtils {
         // Otherwise, just pick one, doesn't matter which.
         existingNote = maybeNotes[0];
       }
-    } else {
-      // Just 1 note
-      existingNote = maybeNotes[0];
-    }
+    } // Else no match in which case we return undefined
```

Comment:
nitpick: This change is unnecceary, if `maybeNotes.length === 0` then `maybeNotes[0] === undefined`.

Reply:
Functionally you are correct, but I'll prefer to explicitly check for certain lengths for readability purposes